compiler/next/lib/parsing/parser-help.h

```diff
@@ -20,10 +20,26 @@
 
 #include <string>
 
+extern const char* yychpl_get_text(yyscan_t scanner);
+
 void yychpl_error(YYLTYPE*       loc,
                   ParserContext* context,
                   const char*    errorMessage) {
-  context->noteError(ParserError(*loc, errorMessage));
+  std::string msg;
+  const char* tokenText = yychpl_get_text(context->scanner);
+  if (strlen(tokenText) > 0) {
+    msg += "near '";
+    msg += tokenText;
+    msg += "'";
+  } else {
+    // Not very helpful, but default parser errors aren't...
+    assert(msg.size() == 0);
+  }
+
+  // Not wasteful, will be collected at the end of the next revision.
+  auto uniqueStrMsg = UniqueString::get(context->context(), msg);
```

Comment:
Why use a UniqueString at all here? Both ParserError and ErrorMessage store a std::string and you already have one of those.